diff --git a/Bluefin/src/main/resources/application-bluefin.properties b/Bluefin/src/main/resources/application-bluefin.properties
index ea3b960d..72a3cb62 100644
--- a/Bluefin/src/main/resources/application-bluefin.properties
+++ b/Bluefin/src/main/resources/application-bluefin.properties
@@ -7,7 +7,7 @@ bluefin.partnerId=gap_hmac
 bluefin.partnerKey=792acf683510feec554dd6719ea9e907
 bluefin.reference=test123
 bluefin.templateRefCreditCard=bc5d057b503920077bdf0bfa785aae6f
-bluefin.templateRefGiftCard=3594ea3b576981294b3c46bb6233c3d6
+bluefin.templateRefGiftCard=e611f64324a4089df653be7d48fea117
 
 # RestTemplate configuration
 restTemplate.connectTimeOut=0
diff --git a/src/main/java/com/gap/customer/vaultservice/util/BluefinUtil.java b/src/main/java/com/gap/customer/vaultservice/util/BluefinUtil.java
index 6500fbe5..674acab6 100644
--- a/src/main/java/com/gap/customer/vaultservice/util/BluefinUtil.java
+++ b/src/main/java/com/gap/customer/vaultservice/util/BluefinUtil.java
@@ -19,6 +19,5 @@ public class BluefinUtil {
             default:
                 return DataType.CreditCardNumber;
         }
-
     }
 }
diff --git a/src/test/java/com/gap/customer/vaultservice/util/BluefinUtilTest.java b/src/test/java/com/gap/customer/vaultservice/util/BluefinUtilTest.java
new file mode 100644
index 00000000..9790a052
--- /dev/null
+++ b/src/test/java/com/gap/customer/vaultservice/util/BluefinUtilTest.java
@@ -0,0 +1,39 @@
+package com.gap.customer.vaultservice.util;
+
+import com.gap.gid.security.model.DataType;
+import org.junit.jupiter.api.Test;
+
+import static com.gap.customer.vaultservice.util.BluefinUtil.getBluefinDatatype;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class BluefinUtilTest {
+    @Test
+    public void shouldReturnCreditCardDatatype() {
+        assertEquals(DataType.CreditCardNumber, getBluefinDatatype("CreditCardNumber"));
+    }
+
+    @Test
+    public void shouldReturnCreditCardExpiryMonthDatatype() {
+        assertEquals(DataType.CreditCardExpiryMonth, getBluefinDatatype("CreditCardExpiryMonth"));
+    }
+
+    @Test
+    public void shouldReturnCreditCardYearDatatype() {
+        assertEquals(DataType.CreditCardExpiryYear, getBluefinDatatype("CreditCardExpiryYear"));
+    }
+
+    @Test
+    public void shouldReturnGiftCardNumberDatatype() {
+        assertEquals(DataType.GiftCardNumber, getBluefinDatatype("GiftCardNumber"));
+    }
+
+    @Test
+    public void shouldReturnGiftCardPinDatatype() {
+        assertEquals(DataType.GiftCardPin, getBluefinDatatype("GiftCardPin"));
+    }
+
+    @Test
+    public void shouldReturnGiftCardTrack2Datatype() {
+        assertEquals(DataType.GiftCardTrack2, getBluefinDatatype("GiftCardTrack2"));
+    }
+}
\ No newline at end of file
