diff --git a/src/functions/other/userInfo/userInfo.ts b/src/functions/other/userInfo/userInfo.ts
index 0051d2c..ee2759d 100644
--- a/src/functions/other/userInfo/userInfo.ts
+++ b/src/functions/other/userInfo/userInfo.ts
@@ -32,9 +32,6 @@ export const handler = async (event: HandlerEvent) => {
   const { onmouuid: queryOnmouuid } = event?.queryStringParameters;
   logger.addContext({ authOnmouuid, queryOnmouuid, scope });
 
-  if (event.requestContext.authorizer) {
-  }
-
   try {
     if (!authOnmouuid) {
       throw new Error("Missing onmouuid from authorizer");
diff --git a/src/services/banking/interface.ts b/src/services/banking/interface.ts
index 3ae5fc9..04b3460 100644
--- a/src/services/banking/interface.ts
+++ b/src/services/banking/interface.ts
@@ -14,7 +14,7 @@ export abstract class BankingServiceInterface {
 export const creditAccountSummarySchema = z.object({
   id: z.string(),
   state: z.string<AccountDetails["state"]>(),
-  createdAt: z.iso.datetime().transform((it) => it && parseISO(it)),
+  createdAt: z.string().transform((it) => it && parseISO(it)),
 });
 
 export type CreditAccountSummary = z.infer<typeof creditAccountSummarySchema>;
diff --git a/src/test-e2e/other/user-info.test.ts b/src/test-e2e/other/user-info.test.ts
index 7a0489b..b2b015b 100644
--- a/src/test-e2e/other/user-info.test.ts
+++ b/src/test-e2e/other/user-info.test.ts
@@ -55,11 +55,7 @@ test(
     );
 
     const userInfoRes = await invokeUserInfo(onmouuid, access_token);
-
-    console.dir(userInfoRes, { depth: 10 });
-
     const userInfoBody = (await userInfoRes.json()) as { loanAccountId: string };
-    console.dir(userInfoBody, { depth: 10 });
     expect(userInfoRes.status).toBe(200);
     expect(userInfoBody).toEqual({ loanAccountId: accountId });
   },
