diff --git a/src/services/banking/interface.ts b/src/services/banking/interface.ts
index a74b26d..a716496 100644
--- a/src/services/banking/interface.ts
+++ b/src/services/banking/interface.ts
@@ -24,7 +24,7 @@ const genericContact = z.object({
 });
 
 export const UserIdMapSchema = z.object({
-  mambuId: z.string('UserIdMap: mambuId not set').optional(),
+  mambuId: z.string("UserIdMap: mambuId not set").optional(),
   creditAccountId: z.string(),
 });
 
diff --git a/src/test-e2e/other/user-info.test.ts b/src/test-e2e/other/user-info.test.ts
index 93280b1..7a0489b 100644
--- a/src/test-e2e/other/user-info.test.ts
+++ b/src/test-e2e/other/user-info.test.ts
@@ -56,10 +56,10 @@ test(
 
     const userInfoRes = await invokeUserInfo(onmouuid, access_token);
 
-    console.dir(userInfoRes, { depth: 10});
+    console.dir(userInfoRes, { depth: 10 });
 
     const userInfoBody = (await userInfoRes.json()) as { loanAccountId: string };
-    console.dir(userInfoBody, { depth: 10});
+    console.dir(userInfoBody, { depth: 10 });
     expect(userInfoRes.status).toBe(200);
     expect(userInfoBody).toEqual({ loanAccountId: accountId });
   },
